teamcity: failed tests on release-2.1: roachtest/acceptance

The following tests appear to have failed:

[#903743](https://teamcity.cockroachdb.com/viewLog.html?buildId=903743):


```
--- FAIL: roachtest/acceptance/gossip/restart-node-one (0.969s)
	test.go:498,gossip.go:321,gossip.go:327,acceptance.go:59: pq: constraint "-rack=0" matches no existing nodes within the cluster - did you enter it correctly?

--- FAIL: roachtest/acceptance/gossip/restart-node-one (0.969s)
	test.go:498,gossip.go:321,gossip.go:327,acceptance.go:59: pq: constraint "-rack=0" matches no existing nodes within the cluster - did you enter it correctly?


```

Please assign, take a look and update the issue accordingly.

--- a/pkg/cmd/roachtest/gossip.go
+++ b/pkg/cmd/roachtest/gossip.go
@@ -28,6 +28,7 @@ import (
 	"unicode"
 
 	"github.com/cockroachdb/cockroach/pkg/gossip"
+	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/httputil"
 	"github.com/cockroachdb/cockroach/pkg/util/retry"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
@@ -322,6 +323,30 @@ func runGossipRestartNodeOne(ctx context.Context, t *test, c *cluster) {
 		}
 	}
 
+	// Wait for gossip to propagate - otherwise attempting to set zone
+	// constraints can fail with an error about how the constraint doesn't match
+	// any nodes in the cluster (#30220).
+	var lastNodeCount int
+	if err := retry.ForDuration(30*time.Second, func() error {
+		const query = `SELECT count(*) FROM crdb_internal.gossip_nodes`
+		var count int
+		if err := db.QueryRow(query).Scan(&count); err != nil {
+			t.Fatal(err)
+		}
+		if count <= 1 {
+			err := errors.Errorf("node 1 still only knows about %d node%s",
+				count, util.Pluralize(int64(count)))
+			if count != lastNodeCount {
+				lastNodeCount = count
+				c.l.Printf("%s\n", err)
+			}
+			return err
+		}
+		return nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+
 	// Evacuate all of the ranges off node 1 with zone config constraints. See
 	// the racks setting specified when the cluster was started.
 	run(`ALTER RANGE default %[1]s CONFIGURE ZONE %[2]s 'constraints: {"-rack=0"}'`)
@@ -330,7 +355,7 @@ func runGossipRestartNodeOne(ctx context.Context, t *test, c *cluster) {
 	run(`ALTER RANGE meta %[1]s CONFIGURE ZONE %[2]s 'constraints: {"-rack=0"}'`)
 	run(`ALTER RANGE liveness %[1]s CONFIGURE ZONE %[2]s 'constraints: {"-rack=0"}'`)
 
-	var lastCount int
+	var lastReplCount int
 	if err := retry.ForDuration(2*time.Minute, func() error {
 		const query = `
 SELECT count(replicas)
@@ -343,8 +368,8 @@ SELECT count(replicas)
 		}
 		if count > 0 {
 			err := errors.Errorf("node 1 still has %d replicas", count)
-			if count != lastCount {
-				lastCount = count
+			if count != lastReplCount {
+				lastReplCount = count
 				c.l.Printf("%s\n", err)
 			}
 			return err

